docs/user/reference/metrics/event.md

```diff
@@ -187,6 +190,9 @@ var snapshot = views::login_opened.test_get_value(None).unwrap();
 assert_eq!(2, snapshot.len());
 let first = &snapshot[0];
 assert_eq!("login_opened", first.name);
+
+let extra = event.extra.unwrap();
+assert_eq!("toolbar", extra.source_of_login);
```

Comment:
Yeah, so apparently you had this more right the first time. Sorry about that. `event.extra` is a `Option<HashMap<String, String>>`